### PR TITLE
Incorporate Eloquent Methods into Model Generation

### PR DESCRIPTION
**Pull request by jschlies**
Incorporate Eloquent Methods into Model Generation

Add code to query default DB and using schema info, generate Eloquent methods
created templates/models/
Added @version phpdoc tag using a timestamp
I'm open to suggestions.

### CODE REVIEW

File: src/Generators/ModelGenerator.php [hunk 1]
```diff
@@ -18,16 +19,55 @@ class ModelGenerator extends BaseGenerator
...
+        $this->table = $this->commandData->dynamicVars['$TABLE_NAME$'];
+        $this->getSchemaManager();
+        $this->getTables();
+        $this->getColumnsPrimaryAndForeignKeysPerTable();
+        $this->getEloquentRules();
     }
 
     public function generate()
     {
-        $templateData = TemplateUtil::getTemplate('model', 'laravel-generator');
```
Comment (jschlies) [inline src/Generators/ModelGenerator.php]: Unsure how you want to handle this as users who use published will need to react to new .stub file structucture

File: src/Generators/ModelGenerator.php [hunk 2]
```diff
@@ -2,6 +2,7 @@
```
Comment (jschlies) [inline src/Generators/ModelGenerator.php]: Thought about creating a stand alone generateEloquent class. Perhaps in next rev.

File: templates/models/hasMany.stub [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
```
Comment (jschlies) [inline templates/models/hasMany.stub]: in next rev, I want to add functions like:
public function attachModel($model)
public function attachModels($model)
public function detachModel($model)
public function detachModels($model)

**Comment by jschlies** [comment]
Mitul,

Here you do. On my fork, I have created jschlies:InfyOmLabs. This will be the branch I will use to move changes via pull requests into InfyOmLabs:develop. Also look at me other branches for other fixes I wish to apply once this big one is through.

**Comment by mitulgolakiya** [comment]
@jschlies Can you put some comments here. what exactly you have developed and how can I generate model with eloquent methods ?

**Comment by jschlies** [comment]
I'll prepare a writeup on this in more detail. In general, to test, do the following:

1. Create a table that has and is referred to my foreign key constraints (unique and non-unique)
1. Generate a model
1. Note the additional helper functions in the model.

**Comment by jschlies** [comment]
Have you been able to generate any models from this. Adding $ELOQUENTFUNCTIONS$ to the model template should do the trick. If correctly generated, you should see several Eloquent helper functions.

**Comment by jschlies** [comment]
We're coming up on 2 months now with this pull request. I'm inclined to continue working on my fork. If any features in my fork interest you, please let me know. Jim

**Comment by mitulgolakiya** [comment]
@jschlies it's not like that I'm not interested in your fork. I am appreciating every fork but this pull request was not a simple one and with lots of commits. so I have to verify each and every piece of code. Also, when we merge something, if should be well documented. so that's another story that I have to write.

And since last month, we followed the practice of develop and master branch, because some users are using our package for production and we can not break it.

Also, I was targeting one major release which I just merged into master on yesterday. so now we have another sprint for next one month for develop branch and I can easily merge each and every PR into develop.

so now I will merge it by today or tomorrow. And really thanks for your contribution.

**Comment by jschlies** [comment]
I think I was too harsh in my comments. Please proceed as you think 
best. If any issues come up, please let me know  and I will address ASAP.

Jim

On 06/14/2016 09:04 PM, Mitul Golakya wrote:
>
> @jschlies <https://github.com/jschlies> it's not like that I'm not 
> interested in your fork. I am appreciating every fork but this pull 
> request was not a simple one and with lots of commits. so I have to 
> verify each and every piece of code. Also, when we merge something, if 
> should be well documented. so that's another story that I have to write.
>
> And since last month, we followed the practice of develop and master 
> branch, because some users are using our package for production and we 
> can not break it.
>
> Also, I was targeting one major release which I just merged into 
> master on yesterday. so now we have another sprint for next one month 
> for develop branch and I can easily merge each and every PR into develop.
>
> so now I will merge it by today or tomorrow. And really thanks for 
> your contribution.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/InfyOmLabs/laravel-generator/pull/173#issuecomment-226082737>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ACGst2HmGiGvM4IzueE8QemOOhjDOldiks5qL3nhgaJpZM4Ip8Hn>.
>

-- 
Jim Schlies
Cell - 831-227-8075

**Comment by phillipmadsen** [comment]
@jschlies , @mitulgolakiya This is a huge step forward and congrats on the accomplishment. I  have been slowly building my own version of this but with my job it is hard to work on alot of the time and yours is alot further along then mine. Now that this is part of the develop branch how would one go about using it in the generator. FieldsFile or commands? Just wondering it this has been figured out. I  will look through the commit to see if it has but if you had the info it would be faster. 

Awesome job guys.

**Pull request merged**